src/pveclib/vec_int512_ppc.h

```diff
@@ -0,0 +1,2167 @@
+ * So the compile does not get much choice for instruction selection.
+ * The PVECLIB coding style does leverage C language vector extensions
+ * to load constants and manage temporary variables.
+ * Using compiler Altivec/VSX built-ins and vector extensions allows
+ * the compiler visibility to and control of these optimizations.
+ *
+ * Internal function calls effectively <I>clobbers</I>
+ * all (34 VSRs) volatile registers.
+ * As the compiler marshals parameters into ABI prescribed VRs
+ * it needs to preserve previous live content for later computation.
+ * Similarly for volatile registers not used for parameter passing
+ * as they are assumed to be clobbered by the called function.
+ * The compiler preserves local live variables before the call by copying
+ * their contents to non-volatile registers or spilling to memory.
+ * This may put more <I>register pressure</I> on the available
+ * non-volatile registers.
+ * Small to medium sized functions often require only a fraction of
+ * the available volatile registers.
+ * In this case, in-lining the function avoids the disruptive volatile
+ * register clobber and allows better overall register allocation.
+ * So there is a strong incentive to in-line local/static functions.
+ *
+ * These compiler optimizations are not independent processes.
+ * For example specific VSX instruction can access all 64 VSRs,
+ * others are restricted to the 32 VRs
+ * (like vector integer instructions).
+ * So the compiler prioritizes VRs (the higher 32 VSRs)
+ * for allocation to vector integer computation.
+ * While the lower 32 VSRs can be used for logical/permute operations
+ * and as a <I>level 1</I> spill area for VRs.
+ * These restrictions combined with code size/complexity can increase
+ * <I>register pressure</I> to the point the compiler is forced to
+ * spill active (or live) vector registers to secondary storage.
+ * This secondary storage can be:
+ * - other architected registers that are available for direct
+ * transfer but not usable in the computation.
+ * - Local variables allocated on the stack
+ * - Compiler temporaries allocated on the stack.
+ *
+ * Instruction scheduling can increase register pressure by moving
+ * (reordering) instructions.
+ * This is more prevalent when there are large differences in
+ * instruction latency in the code stream.
+ * For example moving independent / long latency instructions earlier
+ * and dependent / short latency instructions later.
+ * This tends to increase the distance between the instruction that
+ * sets a register result and the next instruction the uses that
+ * result in its computation.
+ * The distance between a registers set and use is called the
+ * <I>live range</I>.
+ * This also tends to increase the number of concurrently active
+ * and overlapping live ranges.
+ *
+ * For this specific (multiple precision integer multiply) example,
+ * integer multiple and add/carry/extend instructions predominate.
```

Comment:
nit: `multiply`

Reply:
No I mean multiple 

Does "For this specific (multi-precision integer multiply) example," work better?